samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/author/SavePublishedSettingsListener.java

```diff
@@ -563,17 +563,21 @@ private void setPublishedSettings(PublishedAssessmentSettingsBean assessmentSett
 		control.setStartDate(assessmentSettings.getStartDate());
 		control.setDueDate(assessmentSettings.getDueDate());
 
-		if (assessmentSettings.getLateHandling() != null) {
-			control.setLateHandling(new Integer(assessmentSettings.getLateHandling()));
+		Integer lateHandling = assessmentSettings.getLateHandling() != null ? new Integer(assessmentSettings.getLateHandling()) : -1;
+		if (lateHandling > 0) {
+			control.setLateHandling(lateHandling);
 		}
 
-		if (retractNow)
-		{
+		if (retractNow && lateHandling.equals(AssessmentAccessControl.ACCEPT_LATE_SUBMISSION)) {
 			control.setRetractDate(new Date());
+			if (assessmentSettings.getDueDate() != null && assessmentSettings.getDueDate().after(new Date())) {
+				control.setDueDate(new Date());
+			}
```

Comment:
So if they use a retract date AND they have a due date, then set due date to NOW if it's in the future